app/models/username_validator.rb

```diff
@@ -152,6 +155,12 @@ def username_does_not_end_with_confusing_suffix?
     end
   end
 
+  def username_plugin_validations
+    return unless errors.empty?
+
+    DiscoursePluginRegistry.apply_modifier(:username_validator_extras, self.errors, self)
```

Comment:
why extras?

Reply:
It "felt" correct writing the PR, checking it out 3 hours later, maybe `username_validations` feels better, reading it as `plugin apply modifiers on username_validations`

Reply:
Would rename the s I think, just `username_validation`